scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/EDFPolicyExtended.java

```diff
@@ -65,24 +65,38 @@
 
     private static final Set<JobId> alreadyNotified = new HashSet<>();
 
-    private synchronized void logIfJobWillMissItsDeadline(List<JobDescriptor> jobs) {
+    private synchronized void logMostLikelyMissedJobs(List<JobDescriptor> jobs) {
         Date now = new Date();
         jobs.stream()
             .map(jobDescriptor -> ((JobDescriptorImpl) jobDescriptor).getInternal())
             .filter(job -> job.getJobDeadline().isPresent())
-            .filter(job -> job.getJobInfo().getStartTime() >= 0)
+            .filter(job -> !job.getJobInfo().isStarted())
             .filter(job -> {
-                Date deadline = getEffectiveDeadline(job);
-                Date expeFinish = getEffectiveExpectedExecutionTime(job, now);
-                return deadline.compareTo(expeFinish) < 0;
+                Date deadline = getDeadlineAsAbsoluteDate(job);
+                Date finishingTime = getFinishingTimeIfJobStartedNow(job, now);
+                return deadline.compareTo(finishingTime) < 0;
             })
             .forEach(job -> {
                 if (!alreadyNotified.contains(job.getId())) {
-                    LOGGER.warn(String.format("Job[id=%s] might miss its deadline (expected finish: %s after deadline: %s)",
-                                              job.getId().value(),
-                                              getEffectiveExpectedExecutionTime(job, now),
-                                              getEffectiveDeadline(job)));
                     alreadyNotified.add(job.getId());
+                    final Date deadline = getDeadlineAsAbsoluteDate(job);
+                    final Date finishingTime = getFinishingTimeIfJobStartedNow(job, now);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "finishingTime" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)